common/src/main/java/com/wynntils/utils/render/MapRenderer.java

```diff
@@ -146,6 +152,250 @@ public static void renderCursor(
         }
     }
 
+    public static void renderLootrunLine(
+            LootrunInstance lootrun,
+            float lootrunWidth,
+            float outlineWidth,
+            PoseStack poseStack,
+            float centerX,
+            float centerZ,
+            float mapTextureX,
+            float mapTextureZ,
+            float currentZoom,
+            int lootrunColor,
+            int outlineColor) {
+        if (lootrun.simplifiedPath().size() < 3) return;
+
+        BufferBuilder bufferBuilder = Tesselator.getInstance().getBuilder();
+        bufferBuilder.begin(VertexFormat.Mode.TRIANGLES, DefaultVertexFormat.POSITION_COLOR);
+        RenderSystem.setShader(GameRenderer::getPositionColorShader);
+        RenderSystem.setShaderColor(1.0f, 1.0f, 1.0f, 1.0f);
+        RenderSystem.disableCull();
+
+        List<Vector3f> points = new ArrayList<>();
+
+        List<Vector3f> middlePoints = new ArrayList<>();
+
+        for (Vector3f point : lootrun.simplifiedPath()) {
+            points.add(new Vector3f(
+                    getRenderX((int) point.x(), mapTextureX, centerX, currentZoom),
+                    getRenderZ((int) point.z(), mapTextureZ, centerZ, currentZoom),
+                    0));
+        }
+
+        Vector3f last = null;
+        List<Vector3f> filteredPoints = new ArrayList<>();
+        for (Vector3f point : points) {
```

Comment:
I'm trying to make sense of this. First you setup a list `points`, then you discard some of them? Can't you combine this, so you only add the points you really need to the `points` list in the first place?